tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -6155,14 +6155,23 @@ bool HLSLExternalSource::MatchArguments(
       }
     }
 
+    ASTContext &actx = m_sema->getASTContext();
     // Usage
     if (pIntrinsicArg->qwUsage & AR_QUAL_OUT) {
-      if (pCallArg->getType().isConstQualified()) {
+      if (pType.isConstant(actx)) {
         // Can't use a const type in an out or inout parameter.
         badArgIdx = std::min(badArgIdx, iArg);
       }
     }
 
+    // Catch invalid atomic dest parameters
+    if (iArg == kAtomicDstOperandIdx &&
+        IsAtomicOperation(static_cast<IntrinsicOp>(pIntrinsic->Op))) {
+      // bitfield error is confusing
```

Comment:
Still confusing?  Any suggestion to improve it?

Reply:
Well it was non-existent before, so it's less confusing now. It's appears as an overload mismatch that says `note: candidate function not viable: no known conversion from 'uint' to 'unsigned int &' for 1st argument`, which is confusing, but the mechanism doesn't have a mismatch option for bitfields specifically.

Reply:
I see, it's better than nothing!  I guess I was just confused at what this comment in the code was trying to tell the reader of this code.